src/transaction_builder.h

```diff
@@ -49,6 +50,37 @@ struct OutputDescriptionInfo {
     std::optional<OutputDescription> Build(void* ctx);
 };
 
+struct JSDescriptionInfo {
```

Comment:
Something I don't understand about the way this is factored is that values of this struct type are never constructed and passed around as values; instead they're immediately used to build the JSDescription. Why have the data structure at all?

Reply:
I made an explicit datastructure for parity with the other similar structures. In hindsight there isn't much benefit to it (other than keeing the deterministic and randomized builders as closely-coupled as they were as `JSDescription` constructors), since the logic for building Sprout transactions is so horribly complex compared to Sapling.